browser.js

```diff
@@ -190,11 +253,11 @@ Require.loadScript = function (location) {
 Require.ScriptLoader = function (config) {
     var hash = config.packageDescription.hash;
     return function (location, module) {
-        return Promise.fcall(function () {
+        return Promise.try(function () {
 
             // short-cut by predefinition
             if (definitions[hash] && definitions[hash][module.id]) {
-                return definitions[hash][module.id].promise;
+                return definitions[hash][module.id];
```

Comment:
You could just gut the preloading feature. It is evidently not well-tested or maintained. Or write a test to verify it. Mop is capable of creating preload bundle sequences, or you can manually generate similar bundles, and add them to the test suite. They would have to be tested in a script-injection capable environment which is why they aren’t in the Node.js suite. Could be done in PhantomJS.

There’s no way out of implementing defer() unless Bluebird has it already because the promise constructor call site does not close on the state needed to resolve the promise. You could alternately keep the promise and resolver on separate structures, but that is equivalent. https://github.com/petkaantonov/bluebird/blob/master/API.md#deferred-migration